tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -286,5 +295,101 @@ await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile)))
                 }
             }
         }
+
+        private async Task UploadTestRunBlobsAsync(string account, Build build)
+        {
+            try
+            {
+                var continuationToken = string.Empty;
+                var buildIds = new[] { build.Id };
+                var minLastUpdatedDate = build.QueueTime.Value.AddHours(-1);
+                var maxLastUpdatedDate = build.FinishTime.Value.AddHours(1);
+
+                do
+                {
+                    var page = await testResultsClient.QueryTestRunsAsync2(
+                        build.Project.Id,
+                        minLastUpdatedDate,
+                        maxLastUpdatedDate,
+                        continuationToken: continuationToken,
+                        buildIds: buildIds
+                    );
+
+                    foreach (var testRun in page)
+                    {
+                        await UploadTestRunBlobAsync(account, build, testRun);
+                    }
+
+                    continuationToken = page.ContinuationToken;
+                } while (!string.IsNullOrEmpty(continuationToken));
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError(ex, "Error processing test runs for build {BuildId}", build.Id);
+                throw;
+            }
+        }
+
+        private async Task UploadTestRunBlobAsync(string account, Build build, TestRun testRun)
+        {
+            try
+            {
+                var blobPath = $"{build.Project.Name}/{testRun.CompletedDate:yyyy/MM/dd}/{testRun.Id}.jsonl";
```

Comment:
Is `jsonl` intentional?

Reply:
It's for Json Lines format.  The file isn't valid json, but each line is:  https://jsonlines.org/

Reply:
Cool, didn't know about jsonl!